cedar-policy/src/api.rs

```diff
@@ -626,6 +626,19 @@ impl Entities {
     }
 }
 
+type ConvertInternToApiEntity = fn(ast::Entity) -> Entity;
+
+impl IntoIterator for Entities {
+    type Item = Entity;
+
+    type IntoIter =
```

Comment:
I think this type is too complicated, if not idiomatic. IMO, we should follow what Rust standard library does. For instance, `HashMap`'s `IntoIter` type is a dedicated type [`std::collections::hash_map::IntoIter`](https://doc.rust-lang.org/std/collections/hash_map/struct.IntoIter.html). We should at least use an instantiation of this type instead.

Reply:
This is a good point. IIUC, this type becomes part of our public API, so if we don't wrap it would be a potentially braking change to make it return something other `std::iter::Map`

Reply:
Looks good. One tiny idiomatic change, can you rename it to `IntoIter` (like collections do in the standard library)

Reply:
Wouldn't also mean moving it into a module `mod entities { }` so that it's `entities::IntoIter`?